### ISSUE TITLE
`SQLAlchemy` optional dependency

### ISSUE THREAD

**Issue by sbrugman** [reata/sqllineage#624]
**Is your feature request related to a problem? Please describe.**
v1.5.0 introduced a [dependency on SQLAlchemy `>= 2.0.0`](https://github.com/reata/sqllineage/commit/337cc4a11c5776412affdcf5bd8536eea300a1c2). Unfortunately, this version conflicts which what recent version of Airflow require (<2).
I'm not using the SQLAlchemy-based metadata provider.

**Describe the solution you'd like**
The SQLAlchemy could be "feature-gated". 
In `setup.py` we could move the dependency to `extras_require`.
The user can then install this dependency using `pip install sqllineage[sqlalchemy]` or similar.
The core dependency is even lighter!

`sqllineage/core/metadata/sqlalchemy.py` should only be imported when used, so that we get no import errors unless the module is directly imported.

**Describe alternatives you've considered**
-

**Additional context**
Upgrading Airflow is not an option since another team is responsible.

**Comment by reata** [comment]
This decision is made intentionally to not support sqlalchemy v1.x (and as a result, making people suffer a bit).

Airflow and Superset still pin to sqlalchemy v1.x due to the fact that many 3rd party sqlalchemy dialect implementation still uses v1.x. Considering the maintenance effort, I don't want to build a compatible layer regarding sqlalchemy v1.x and v2.x API, though that would be more user-friendly. (this looks like what Airflow will do, see https://github.com/apache/airflow/issues/28723)

At the same time, even though sqlalchemy v2.x is marked as dependency, we're coding in an OOP way that if you don't use sqlalchemy metadata provider, your code will work even with sqlalchemy v1.x installed. (pip might remind you of incompatible dependency, but that's fine). 

Let me know if this works before we discuss on the 'optional' approach.

**Comment by sbrugman** [comment]
I agree with the decision not to support sqlalchemy v1.x, no problem there.

When the build pipeline includes a `pip compile` (or similar) step like we have, there is no resolution in the current setting. I still think the 'optional' approach is closest to the desired behaviour: if the user would like the sqlalchemy metadata provider, then the version should be >= v2.x

**Comment by cheunhong** [comment]
Agree on this

**Comment by rubytobi** [comment]
We are using sqllineage purely as a Python package, not the UI functionality. Given other requirements, we are still stuck with sqlalchemy 1.x as well, which lead us to forking the repository to drop the sqlalchemy requirement for a clean install via git.

Would be great to have sqlalchemy as an optional dependency for those who do want to use it, but not force others into it as well.

**Comment by reata** [comment]
> Thanks for looking into this. I agree, just dropping it in a minor release would be unexpected and break common consensus on SemVer.
> 
> How would you feel about a v2 or v1.7 release?

Let's not take the easy route. We're not using sqlalchemy feature heavily as airflow does. It shouldn't be much effort to make it compatible with v1 and v2.
